EventFilter/EcalRawToDigi/interface/EcalRawDataRecovery.h

```diff
@@ -0,0 +1,83 @@
+#ifndef ECALRAWDATARECOVERY_H
+#define ECALRAWDATARECOVERY_H
+
+//Emacs settings:-*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil -*-
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <memory>
+#include <inttypes.h>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "DataFormats/EcalRawData/interface/EcalRawDataCollections.h"
+
+//forward declaration
+class FEDRawDataCollection;
+
+/** EcalRawDataRecover class
+ *
+ * CMSSW module to fix identified ECAL data corruption. Current implementation
+ * fixes the problem of run 274157 (see fixRun274157). The code can be extended
+ * to deal with data issue that would be found in new runs.
+ * Original author: Ph. Gras CEA/Saclay
+ */
+class EcalRawDataRecovery : public edm::EDProducer {
```

Comment:
new modules have to be MT-aware https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes